### PR TITLE
Contribute to app-i18n/fcitx-bamboo

### PR DESCRIPTION
**Pull request by vuongtuha**

### CODE REVIEW

File: app-i18n/fcitx-bamboo/metadata.xml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE pkgmetadata SYSTEM "https://www.gentoo.org/dtd/metadata.dtd">
+<pkgmetadata>
+	<maintainer type="person">
+		<email>tuhavuong@gmail.com</email>
+		<name>Vương Tử Hà</name>
+	</maintainer>
+	<use>
+		<flag name="qt">Enable Qt for GUI configuration</flag>
```
Comment (Universebenzene) [inline app-i18n/fcitx-bamboo/metadata.xml]: `qt` flag is not appeared in your ebuild. No need to write this in metadata

However you defined a `go` flag in your ebuild without description. And I don't understand how this flag works

File: app-i18n/fcitx-bamboo/fcitx-bamboo-1.0.1.ebuild [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+src_configure() {
+	local mycmakeargs=(
+		-DCMAKE_INSTALL_LIBDIR="${EPREFIX}/usr/$(get_libdir)"
+	)
+	cmake_src_configure
+}
+
+pkg_postinst() {
+	xdg_pkg_postinst
```
Comment (Universebenzene) [inline app-i18n/fcitx-bamboo/fcitx-bamboo-1.0.1.ebuild]: `xdg-utils.eclass` is not enough for this function. Use `xdg.eclass` instead

Also `xdg.eclass` will do this automatically. You don't need to write this out.

**Review by Universebenzene — CHANGES_REQUESTED** [review]

File: app-i18n/fcitx-bamboo/fcitx-bamboo-1.0.1.ebuild [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+"
+RDEPEND="${DEPEND}"
+BDEPEND="
+	dev-lang/go
+	kde-frameworks/extra-cmake-modules
+"
+
+src_configure() {
+	local mycmakeargs=(
+		-DCMAKE_INSTALL_LIBDIR="${EPREFIX}/usr/$(get_libdir)"
```
Comment (Universebenzene) [inline app-i18n/fcitx-bamboo/fcitx-bamboo-1.0.1.ebuild]: This should be the default behavior of  `cmake.eclass`. I think you don't need to write the `src_configure` phase out

**Comment by vuongtuha** [comment]
Thanks for your review. 1st of all, before creating this ebuild, I had to
wonder about how should I force the user to USE="+qt". I asked even
upstream that could I make it build without go-lang.
Then even before pulling this request, I hoped that someone could test my
build. And would you mind if I ask "is Telegram channel Gentoo-zh the place
to share about this? ". Or are you in this channel?
Sorry I don't know Chinese, but 1st I think I need somewhere to get the
answer to my question. Ofc, you decide what is suitable for your repo, it's
your permission. Maybe I need others' help to finish this ebuild.  If you
feel fine about this, just let me know. Thanks!

**Comment by Universebenzene** [comment]
@vuongtuha 
> , before creating this ebuild, I had to wonder about how should I force the user to USE="+qt".

`qt` flag is not appeared in any conditions in your ebuild. Whether people enabled this flag or not will do nothing for your package.



> I asked even upstream that could I make it build without go-lang.

If this package really needs `dev-lang/go` for compiling, just write it in `BDEPEND` is enough. Defining a USE flag without any condition will do nothing.

**Comment by st0nie** [comment]
@github-actions[bot] 改成inherit cmake xdg，然后go的依赖先去掉吧。pkg_postinst() 一整段也可以删掉

**Comment by Universebenzene** [comment]
@ston-afk 以及metadata里的use desc也应该删了

**Comment by st0nie** [comment]
@Universebenzene 还得rebase一下（

**Comment by st0nie** [comment]
@github-actions[bot] DEPEND不应该包含configtool
src_configure()和pkg_postinst()可以整个都删掉了，这是eclass的默认行为，不需要你去写

**Pull request closed without merging**